.github/workflows/CI.yml

```diff
@@ -17,45 +14,78 @@ on:
   workflow_dispatch:
 
 jobs:
-  build:
+  license-check:
+    name: Verify repository license compliance
     runs-on: ubuntu-latest
     container: khronosgroup/docker-images:asciidoctor-spec
 
     steps:
-      #  Unfortunately, asciidoctor-pdf gets pathname-specific errors
-      # building under the usual $GITHUB_WORKSPACE (/__w). As a workaround,
-      # generate the outputs in /tmp.
+      - uses: actions/checkout@v3
+      - name: REUSE license checker
+        run: reuse lint
 
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-
-      # REUSE license checker
-      - name: license-check
-        run: |
-          reuse lint
+  terminology-check:
+    name: Run various checker scripts on the spec and XML sources
+    runs-on: ubuntu-latest
+    container: khronosgroup/docker-images:asciidoctor-spec
 
-      # Run various checker scripts on the spec and XML sources
-      - name: terminology_check
+    steps:
+      - uses: actions/checkout@v3
+      - name: Internal self-test of the check_spec_links script
+        run: py.test-3 test*.py
+        working-directory: scripts
+      - name: Generate a summary of problems for CI logs (if any)
         run: |
-          # Internal self-test of the check_spec_links script
-          ( cd scripts && py.test-3 test*.py )
           mkdir -p gen/out/checks
           scripts/check_spec_links.py --html=gen/out/checks/problems.html > /dev/null || true
           make allchecks
 
-      # Build spec targets
-      - name: spec-generate
-        run: |
-          ./makeSpec -clean -spec all QUIET= -j${nproc} -Otarget manhtmlpages validusage styleguide registry chunked html
-          ./makeSpec -clean -spec core -genpath gencore QUIET= -j${nproc} -Otarget chunked html
-          ( cd xml && make validate install test )
+  spec-generate:
+    name: Build the vulkan specification and generate any associated files (such as vulkan.h)
+    runs-on: ubuntu-latest
+    container: khronosgroup/docker-images:asciidoctor-spec
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build the actual spec (both chunked and single-page HTML), and other common targets
+        run: ./makeSpec -clean -spec all QUIET= -j${nproc} -Otarget manhtmlpages validusage styleguide registry chunked html
+      - name: Build the core-only spec, to try and catch ifdef errors in extension markup
```

Comment:
This is looking good - thanks very much for taking a run at this. BTW the two makeSpec builds are entirely independent, and could be done separately if you want to add that. The first run takes much longer than the second since it's building 1500-odd HTML reference pages as well as the spec, but even so the second adds 17s+ to the overall runtime. I am a little surprised at how slow the first build is, though - maybe we don't get many cores in the build environment. I usually build with -j12 or -j24 on my own 6-core machine.

I noticed you split the multiple-line run: | actions into separate name:/run: for each command line. Is that just for clarity and to pull the comments up into metadata in the CI script, or is there another benefit?

Reply:
> This is looking good

Thanks!

> BTW the two makeSpec builds are entirely independent, and could be done separately if you want to add that.

Let's separate them and see the results :)

> but even so the second adds 17s+ to the overall runtime.

It'll probably cost more cycles total when running in parallel, but should save on combined run length.

> I am a little surprised at how slow the first build is, though - maybe we don't get many cores in the build environment. I usually build with -j12 or -j24 on my own 6-core machine.

https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources It's only running on a 2-core VM, no ThreadRippers :/

> I noticed you split the multiple-line run: | actions into separate name:/run: for each command line. Is that just for clarity and to pull the comments up into metadata in the CI script, or is there another benefit?

Yeah, to pull comments into the metadata and show them in the job run overview. Otherwise, when a single step fails you're might be left wondering which of the lines in the script are at fault here (though, it's usually easy to guess). There was a thing in the past where these scripts were running without `set -x` and only the exit code of the last line was considered for job fail/success status.

Reply:
> > but even so the second adds 17s+ to the overall runtime.
> 
> It'll probably cost more cycles total when running in parallel, but should save on combined run length.

As expected this job just ran at ±20s but takes an extra ±40s for the container setup. Strangely enough total runtime got decreased to 9m9s, probably quiet time :sweat_smile: